crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java

```diff
@@ -349,11 +347,10 @@ public CRUDFindResponse find(CRUDOperationContext ctx,
                 ctx.addError(Error.get(MongoCrudConstants.ERR_NO_ACCESS, "find:" + ctx.getEntityName()));
             }
         } catch (Error e) {
-            // rethrow lightblue error
-            throw e;
+            ctx.addError(e);
```

Comment:
Yes, all Error's are propagated back to client. Anything that is internal
to the system is throw using regular exceptions, and if they propagate,
they are eventually put into an Error.

I'll remove the redundant logs, and add logging to Error.get


On Thu, Jul 9, 2015 at 1:14 PM, Marek <notifications@github.com> wrote:

> In
> crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/130#discussion_r34292117>
> :
>
> > @@ -349,11 +347,10 @@ public CRUDFindResponse find(CRUDOperationContext ctx,
> >                  ctx.addError(Error.get(MongoCrudConstants.ERR_NO_ACCESS, "find:" + ctx.getEntityName()));
> >              }
> >          } catch (Error e) {
> > -            // rethrow lightblue error
> > -            throw e;
> > +            ctx.addError(e);
>
> Are Error.get always called when an error happens? If so and they are all
> supposed to log, then statements
> LOGGER.error(e.getMessage(), e);
> in catch are redundant.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/130/files#r34292117>
> .
>